frontend/pages/admin/UserManagementPage/components/UserForm/UserForm.tsx

```diff
@@ -210,84 +248,45 @@ const UserForm = ({
   };
 
   const validate = (): boolean => {
-    if (!validatePresence(formData.name)) {
-      setErrors({
-        ...errors,
-        name: "Name field must be completed",
-      });
+    const newErrors: IUserFormErrors = {};
 
-      return false;
+    if (!validatePresence(formData.name)) {
+      newErrors.name = "Name field must be completed";
     }
-
     if (!validatePresence(formData.email)) {
-      setErrors({
-        ...errors,
-        email: "Email field must be completed",
-      });
-
-      return false;
-    }
-
-    if (!validEmail(formData.email)) {
-      setErrors({
-        ...errors,
-        email: `${formData.email} is not a valid email`,
-      });
-
-      return false;
-    }
-
-    if (
-      !isNewUser &&
-      !isInvitePending &&
-      formData.password &&
-      !validPassword(formData.password) &&
-      !formData.sso_enabled
-    ) {
-      setErrors({
-        ...errors,
-        password: "Password must meet the criteria below",
-      });
-
-      return false;
+      newErrors.email = "Email field must be completed";
+    } else if (!validEmail(formData.email)) {
+      newErrors.email = `${formData.email} is not a valid email`;
     }
-
+    // Only test password on new created user (not invited user) or if sso not enabled
     if (
       isNewUser &&
       formData.newUserType === NewUserType.AdminCreated &&
       !formData.sso_enabled
     ) {
-      if (!validatePresence(formData.password)) {
-        setErrors({
-          ...errors,
-          password: "Password field must be completed",
-        });
-
-        return false;
+      if (formData.password !== null && !validPassword(formData.password)) {
+        newErrors.password = "Password must meet the criteria below";
       }
-      if (!validPassword(formData.password)) {
-        setErrors({
-          ...errors,
-          password: "Password must meet the criteria below",
-        });
-
-        return false;
+      if (!validatePresence(formData.password)) {
+        newErrors.password = "Password field must be completed";
       }
     }
 
+    setErrors(newErrors);
+
     if (!formData.global_role && !formData.teams.length) {
       renderFlash("error", `Please select at least one team for this user.`);
       return false;
     }
-
-    return true;
+    return Object.keys(newErrors).length === 0;
   };
 
   const onFormSubmit = (evt: FormEvent): void => {
     evt.preventDefault();
+
     const valid = validate();
```

Comment:
Good opportunity to apply our (validation
pattern)[https://github.com/fleetdm/fleet/blob/163be5deee0a87dd95199b2356af047fcfd35bf6/frontend/docs/patterns.md#data-validation],
i.e., `validate` should be a pure function that returns errors (then you can set state right here
with them, instead of within `validate`)

Reply:
great suggestion, can iterate later this week if time permits